gqlalchemy/query_builders/memgraph_query_builder.py

```diff
@@ -79,8 +80,112 @@ def load_csv(self, path: str, header: bool, row: str) -> "DeclarativeBase":
 
         return self
 
+    def construct_subgraph_query(
+        self,
+        node_labels: Optional[Union[str, Iterable[str]]] = None,
+        relationship_types: Optional[Union[str, Iterable[str]]] = None,
+        relationship_direction: Optional[RelationshipDirection] = RelationshipDirection.RIGHT,
+    ) -> str:
+        """Constructs a MATCH query defining a subgraph using one or two node labels and relationship types.
+
+        Args:
+            node_labels: If string, used for both sides of the relationship, if list of two, used in order.
+            relationship_types: A string or iterable of types of relationships used in the subgraph.
+            relationship_direction: Enum representing direction.
+
+        Returns:
+            a string representing a MATCH query for a path with given node labels and relationship types.
+        """
+        if not node_labels or isinstance(node_labels, str):
+            first_node = second_node = CypherNode(node_labels)
+        else:
+            if len(node_labels) > 2:
+                raise ValueError("only two node labels are permitted, one for outgoing node and one for incoming node.")
+            first_node = CypherNode(node_labels[0])
+            second_node = CypherNode(node_labels[1])
+
+        query = f" MATCH p={first_node}{CypherRelationship(relationship_types, relationship_direction)}{second_node}"
+
+        return query
+
+    def call(
+        self,
+        procedure: str,
+        arguments: Optional[Union[str, Tuple[Union[str, int, float]]]] = None,
+        node_labels: Optional[Union[str, Iterable[str]]] = None,
+        relationship_types: Optional[Union[str, Iterable[str]]] = None,
+        relationship_direction: Optional[RelationshipDirection] = RelationshipDirection.RIGHT,
+        subgraph_query: str = None,
+    ) -> "DeclarativeBase":
```

Comment:
maybe it is better to use always `RelationshipDirection.RIGHT` to simplify it? I'm not sure if we need both since we can only query a subgraph of max 2 entities here

Reply:
Since it is a default value, I'd keep it in, for user flexibility. If we hardcode it, it might be more confusing in some sense, because with this users only need to look at the argument list, and have everything at one place.

But I'm not stubborn about this, let me know if I should change it.

Reply:
okay if there is a default value then fine, it's okay